pandas/io/parsers/base_parser.py

```diff
@@ -1137,6 +1137,9 @@ def unpack_if_single_element(arg):
         return arg
 
     def converter(*date_cols, col: Hashable):
+        if len(date_cols) == 1 and date_cols[0].dtype.kind in "Mm":
```

Comment:
Not sure if I understood your comment in the OP correctly, but does this also fix `parse_dates` for numpy dtypes too?

Reply:
Only NumPy dtypes as far as I know, we fixed this for Arrow dtypes a couple of weeks ago.